client/js/render.js

```diff
@@ -72,6 +75,12 @@ function buildChatMessage(data) {
 	const msg = $(templates[template](data.msg));
 	const text = msg.find(".text");
 
+	if (data.msg.previews.length) {
+		data.msg.previews.forEach((preview) => {
```

Comment:
Essentially, the following:

```js
data.msg.previews.forEach((preview) => renderPreview(preview, msg));
```

translates to:

```js
data.msg.previews.forEach((preview) => {
  return renderPreview(preview, msg);
});
```

and not:

```js
data.msg.previews.forEach((preview) => {
  renderPreview(preview, msg);
});
```

That extra `return` is unwanted.